[Doc] Add docs for resource usage stats (#5408)
opensearch-project/documentation-website: _api-reference/nodes-apis/nodes-stats.md

script_cache | Statistics about script cache.
indexing_pressure | Statistics about the node's indexing pressure.
shard_indexing_pressure | Statistics about shard indexing pressure.

> resource_usage_stats | Resource usage statistics of nodes, such as CPU and JVM

Reviewer: ```suggestion
resource_usage_stats | Node-level resource usage statistics, such as CPU and JVM memory.
```

[indexing_pressure](#indexing_pressure) | Object | Statistics related to the node's indexing pressure.
[shard_indexing_pressure](#shard_indexing_pressure) | Object | Statistics related to indexing pressure at the shard level.
[search_backpressure]({{site.url}}{{site.baseurl}}/opensearch/search-backpressure#search-backpressure-stats-api) | Object | Statistics related to search backpressure.

> [resource_usage_stats](#resource_usage_stats) | Object | Statistics related to resource usage stats for the node.

Reviewer: ```suggestion
[resource_usage_stats](#resource_usage_stats) | Object | Statistics related to resource usage for the node.
```

enabled | Boolean | Specifies whether the shard indexing pressure feature is turned on for the node.
enforced | Boolean | If true, the shard indexing pressure runs in enforced mode (there are rejections). If false, the shard indexing pressure runs in shadow mode (there are no rejections, but statistics are recorded and can be retrieved in the `total_rejections_breakup_shadow_mode` object). Only applicable if shard indexing pressure is enabled. 

### `resource_usage_stats`
The `resource_usage_stats` object contains the resource usage statistics. Each entry is specified by the node ID and has the following properties.

Field | Field type | Description
:--- |:-----------| :---
timestamp | Integer    | The last refresh time for the resource usage statistics, in milliseconds since the epoch.

> cpu_utilization_percent | Float      | Statistics for the average CPU usage of OpenSearch process in the period configured in the setting `node.resource.tracker.global_cpu_usage.window_duration`.

Reviewer 2: Can we move field type to `double` instead of `float`.

> cpu_utilization_percent | Float      | Statistics for the average CPU usage of OpenSearch process in the period configured in the setting `node.resource.tracker.global_cpu_usage.window_duration`.

Reviewer: ```suggestion
cpu_utilization_percent | Float      | Statistics for the average CPU usage of OpenSearch process within the time period configured in the `node.resource.tracker.global_cpu_usage.window_duration` setting.
```

> memory_utilization_percent | Float      | Statistics for the JVM memory usage on the node in the period configured in the setting `node.resource.tracker.global_jvmmp.window_duration`

Reviewer: ```suggestion
memory_utilization_percent | Float      | The node JVM memory usage statistics within the time period configured in the `node.resource.tracker.global_jvmmp.window_duration` setting.
```
